File: library/src/main/java/com/library/management/service/interfaces/NotificationService.java
package com.library.management.service.interfaces;

import com.library.management.dto.response.NotificationResponse;

import java.util.List;

public interface NotificationService {

    NotificationResponse createNotification(
            Long userId,
            String title,
            String message,
            String type
    );

    // Student Notification
    void notifyStudent(
            Long studentId,
            String title,
            String message,
            String type
    );

    // Admin + Librarian Notification
    void notifyAdmins(
            String title,
            String message,
            String type
    );

    List<NotificationResponse> getNotifications(Long userId);

    List<NotificationResponse> getUnreadNotifications(Long userId);

    long getUnreadCount(Long userId);

    void markAsRead(Long notificationId);

    void deleteNotification(Long notificationId);
    void markAllAsRead(Long userId);

    void deleteAllRead(Long userId);

}

File: library-frontend/src/services/notificationService.js
import api from "../api/axiosConfig";

export const getNotifications = async (userId) => {

    const response = await api.get(`/notifications/${userId}`);

    return response.data;

};

export const getUnreadNotifications = async (userId) => {

    const response = await api.get(`/notifications/unread/${userId}`);

    return response.data;

};

export const getUnreadCount = async (userId) => {

    const response = await api.get(`/notifications/count/${userId}`);

    return response.data;

};

export const markAsRead = async (notificationId) => {

    await api.put(`/notifications/${notificationId}/read`);

};

export const deleteNotification = async (notificationId) => {

    await api.delete(`/notifications/${notificationId}`);

};
export const markAllAsRead = async (userId) => {

    await api.put(`/notifications/read-all/${userId}`);

};

export const deleteAllRead = async (userId) => {

    await api.delete(`/notifications/read/${userId}`);

};